airbyte-cdk/java/airbyte-cdk/typing-deduping/src/testFixtures/java/io/airbyte/integrations/base/destination/typing_deduping/BaseSqlGeneratorIntegrationTest.java

```diff
@@ -382,7 +382,7 @@ public void incrementalDedupSameNameNamespace() throws Exception {
             }
             """)));
 
-    final String sql = generator.updateTable(stream, "", Optional.empty());
+    final String sql = generator.updateTable(stream, "", Optional.empty(), true);
```

Comment:
Seems like there are no tests for the `false` case... should there be?

Reply:
looked into this - added both options where appropriate, the sql generator tests also sometimes assume that casting errors will be caught but this is probably a misplaced test, these tests should probably be in in the `DefaultTyperDeduperTest` to avoid duplicating this logic for testing purposes.